### PR TITLE
Remove Ent Part 1

### PR DESCRIPTION
**Pull request by LBeernaertProton**
Hide ent implementation behind an interface

### CODE REVIEW

File: db/errors.go [hunk 1]
```diff
@@ -0,0 +1,14 @@
+package db
+
+import "errors"
+
+var ErrNotFound = errors.New("value not found")
+var ErrTransactionFailed = errors.New("value not found")
```
Comment (rlejeune74) [inline db/errors.go:6]: should be something more like "transaction failed"

File: db/ops_message.go [hunk 2]
```diff
@@ -0,0 +1,94 @@
...
+
+	MessageExistsWithRemoteID(ctx context.Context, id imap.MessageID) (bool, error)
+
+	GetMessage(ctx context.Context, id imap.InternalMessageID) (*Message, error)
+
+	GetTotalMessageCount(ctx context.Context) (int, error)
+
+	GetMessageRemoteID(ctx context.Context, id imap.InternalMessageID) (imap.MessageID, error)
+
+	//GetMessageWithRemoteID(ctx context.Context, id imap.MessageID) (*Message, error)
```
Comment (rlejeune74) [inline db/ops_message.go:22]: dead code ?

File: db/errors.go (hunk 1 above)
Comment (LBeernaertProton) [reply db/errors.go]: Good catch. Fixed.

File: db/ops_message.go (hunk 2 above)
Comment (LBeernaertProton) [reply db/ops_message.go]: Yep. Fixed.

**Review by rlejeune74 — APPROVED** [review]

**Pull request merged**